### PR TITLE
Adds glossary entries for 'sos report' and 'sosreport'

### PR DESCRIPTION
**Pull request by bredamc**
Proposed fix for issue #315 (Correct spelling and capitalization for "sosreport").

### CODE REVIEW

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bburt-rh) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:618]:
```suggestion
*Incorrect forms*: sosreport (incorrect except when referring to the command name in RHEL 7 and earlier), SOSreport, SOS report, or any other variation
```
Maybe better to say that it's always incorrect except for this one usage case? WDYt?

Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: Great point, @bburt-rh -- I've made that change, thanks!

**Review by bburt-rh — DISMISSED** [review]

**Comment by bburt-rh** [comment]
lgtm

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (IngridT1) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: @bergerhoffer @bredamc @bburt-rh This language is mostly fine and a good effort. I would have added this edit directly to the text, but I think we'll need to discuss it here first:

- Question 1. Should we say that if you refer to "sosreport" or "sos report", it should always be written as code in a monospaced font? That's how I usually see it in the docs. We probably need to make that explicit.
- Question 2. OpenShift talks about generating a "sosreport" here: https://docs.openshift.com/container-platform/4.13/support/gathering-cluster-data.html#about-sosreport_gathering-cluster-data How do you know if "sos report" or "sos report" is correct if you're not working in RHEL? Would OpenShift 4.13 for sure be on RHEL8 or is it variable? How do we handle "sos report" in that situation?

Thanks,
Ingrid

Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: I updated the PR to split `sos report` and `sosreport` into two separate "Use with caution" entries. 

I also included the word "archive", because I think the main problem area is what to call the file that's created when you run the command, and "archive" indicates that it's an aggregate file.

Let me know what you think!

Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: Per the SSG meeting, I think we wanted to update the entries to remove the word "archive" after, does that sound right?

Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: Yes, and include a mention of the term that is used for the other RHEL version (in each term entry). It's on my list!

**Review by bburt-rh — APPROVED** [review]

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:638]: The use it is marked as "yes", but this uses the "with caution" icon. Can you update the icon for both of these entries?

**Review by bergerhoffer — COMMENTED** [review]
A few pieces of feedback!

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (dfitzmau) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:625]: Is this an example? An example heading or some sort of flag should be provided.

Consider swapping out parentheses for a semicolon as the semicolon provides a better contrast with the transitional phrase that follows.

> To generate an `sos report` archive, run the `sos report` command; that is, the `sos` command with the `report` argument.

**Comment by dfitzmau** [comment]
Should a preview link be made available on each PR?

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (dfitzmau) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:625]: What about generating an `sos report` file? Is this a different entity in comparison to an archive. I think an _example_ heading would clear things here.

Comment (dfitzmau) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:625]: Woth mentioning that on a Linux system one can run the command `man sos` for more info about the command and its options?

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: Removed the references to "adjective".

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: Removed the unnecessary italics :)

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: Changed the "Use it" text to "with caution" to ensure that people use the right term for their RHEL version.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: 1. It's not an example (although it does show the usage!). Moved the sentence to the initial paragraph and removed the parentheses as suggested.
2. Removed all references to _archive_ and _file_ to avoid confusion.
3. I don't think it's necessary to add this info. The point of the entries is to indicate how to write the terms.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:649]: I don't think we need to include a reference to the IBM product? It doesn't add anything to the explanation of how to write the terms "sos report" or "sosreport", and might introduce confusion?

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: @bergerhoffer Do you agree?

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:649]: @bergerhoffer @mportman12 Do you agree?

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: I agree that it's not necessary to add the man command.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:649]: I think we're okay without it. There are probably lots of other things that uses "SOS". and ours is lowercase, and has "report" after it, so there shouldn't be confusion.

**Review by bergerhoffer — DISMISSED** [review]
All of my comments were addressed, but could still use some fresh eyes. 

Marking approved from my side!

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc
Comment (dfitzmau) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:649]: Thanks for confirming. Maybe add "SOS" to the `Incorrect forms` field?

Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:649]: _(Apologies, I click the wrong button and resolved by mistake!)_

And apologies for disagreeing with you again @dfitzmau but I think it's unnecessary to add "SOS" to the "Incorrect forms" field -- as far as I'm aware, that incorrect form has not been used to refer to the sos report. To me, your suggestion is analogous to adding "BASIC" to the "Incorrect forms" field for the "Basic HTTP authentication" entry. But I'll defer to the majority decision :)

Comment (dfitzmau) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:649]: NP. I just spotted some of Red Hat's support articles that use the acronym "SoS". Example: https://access.redhat.com/articles/445443

Sometimes doc Jira requests might include a link to article for supplementary information, so if this article were attached, maybe a writer could fall into the trap of using `SoS` in their content. Please mark the comment as "Resolved" if my comment is still not applicable. Thanks!

Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc:649]: Ah, good spot! I'll add "SoS report" and "SoS Report" to the list of Incorrect forms! Thanks for your patience, @dfitzmau :)

Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc]: Done. I'll resolve this conversation, but feel free to reopen it if you'd like to discuss any of your points further :)

**Review by dfitzmau — APPROVED** [review]

**Review by sbmetz — APPROVED** [review]
LGTM

**Comment by jseseCCS** [comment]
not sure if you still need it, but approved! LGTM :nerd_face:

**Pull request merged**
